client/src/worker_registry/mod.rs

```diff
@@ -58,24 +60,21 @@ impl SlotKey {
 /// This is an inner class to hide the mutex.
 #[derive(Default, Debug)]
 struct SlotManagerImpl {
-    /// Maps keys, i.e., namespace#task_queue, to providers.
-    providers: HashMap<SlotKey, Vec<Box<dyn SlotProvider + Send + Sync>>>,
+    /// Maps keys, i.e., namespace#task_queue, to provider.
+    providers: HashMap<SlotKey, Box<dyn SlotProvider + Send + Sync>>,
     /// Maps ids to keys in `providers`.
     index: HashMap<String, SlotKey>,
-}
-
-fn random_permutation(num_entries: usize) -> Vec<usize> {
-    let mut vec: Vec<usize> = (0..num_entries).collect();
-    vec.shuffle(&mut thread_rng());
-    vec
+    /// Do not warn when ignoring worker registrations.
+    quiet: bool,
```

Comment:
Why add this? I don't think we need this any more than we need to ability to quiet warnings elsewhere in the codebase. Typically they can be silenced with an appropriate filter string in the logging config anyway.

Reply:
In the unit tests I'm using a static SlotManager, and since workers do not properly unregister it could create many unnecessary warn! messages. So that flag just means "mock" the slot manager...

Reply:
It's perfectly fine for unit tests to log warnings. Plus, the logger isn't even turned on for unit tests unless you explicitly add it.

Reply:
Great! Done...